vue/test/components/ripe-pickers/ripe-pickers.test.js

```diff
@@ -1,19 +1,158 @@
+                    },
+                    leather_cbe: {
+                        available: true,
+                        colors: {
+                            black: { available: true },
+                            blue: { available: true },
+                            brown: { available: true },
+                            red: { available: true }
+                        }
+                    },
+                    suede_cbe: {
+                        available: true,
+                        colors: {
+                            blue: { available: true },
+                            fuchsia: { available: true },
+                            red: { available: true }
+                        }
+                    }
+                }
+            },
+            patch: {
+                available: true,
+                materials: {
+                    patch_cbe: {
+                        available: true,
+                        colors: { smiley: { available: true }, egg: { available: true } }
+                    }
+                }
+            }
+        };
+        this.options = {
+            parts: {
+                side: {
+                    color: "black",
+                    material: "crocodile_cbe",
+                    face: "side"
+                },
+                shadow: {
+                    color: "default",
+                    hidden: true,
+                    material: "default"
+                },
+                top0_bottom: {
+                    color: "fuchsia",
+                    face: "side",
+                    material: "suede_cbe"
+                }
             }
-        });
+        };
+    }
+
+    isReady() {}
+    bind() {}
+}
+
+describe("RipePickers", () => {
+    it("should instantiate the component", async () => {
+        const ripeInstance = new MockRipeSdk();
+        const component = await base.getComponent("RipePickers", { props: { ripe: ripeInstance } });
 
         assert.strictEqual(component.emitted("loading").length, 1);
+
+        await component.vm.$forceUpdate();
+        assert.strictEqual(component.find(".select.parts").exists(), true);
+        assert.strictEqual(component.find(".select.materials").exists(), true);
+        assert.strictEqual(component.find(".select.colors").exists(), true);
+    });
+
+    it("should sync selects options", async () => {
+        const ripeInstance = new MockRipeSdk();
+        const component = await base.getComponent("RipePickers", { props: { ripe: ripeInstance } });
+
+        await component.vm.$forceUpdate();
+        assert.strictEqual(component.find(".select.parts").findAll("option").length, 4);
+        assert.strictEqual(component.find(".select.materials").findAll("option").length, 1);
+        assert.strictEqual(component.find(".select.colors").findAll("option").length, 1);
+
+        await component.setData({ selectedPart: "side" });

```

Comment:
Use the component's public API instead. It should be `selectPart("side")` but we will have to settle for `onSelectPartChange` since it's what exists (this is why we should always separate handlers from actual methods).